app/Components/Draggable.js

```diff
@@ -34,19 +35,27 @@ export default function Draggable({
       initialX: x,
       initialY: y,
     };
+    selectedItem = { id: vector.id };
     updateNewVectors(updatedVector);
+    setTargetId(vector.id);
   };
 
   const pan = Gesture.Pan()
     .onChange((event) => {
       handleChange(event);
     })
-    .onFinalize(() => {
-      handleFinalize();
+    .onFinalize((event) => {
+      handleFinalize(event);
```

Comment:
I think you can remove the event since the `handleFinalize` function is not taking any parameters.